obstacle_stop_planner_refine/include/obstacle_stop_planner/one_step_polygon.hpp

```diff
@@ -0,0 +1,159 @@
+    one_step_move_vehicle_corner_points.push_back(
+      cv::Point2d(
+        base_step_pose.position.x + std::cos(yaw) * (-vehicle_info_->rear_overhang_m_) -
+        std::sin(yaw) * (vehicle_info_->vehicle_width_m_ / 2.0 + expand_width),
+        base_step_pose.position.y + std::sin(yaw) * (-vehicle_info_->rear_overhang_m_) +
+        std::cos(yaw) * (vehicle_info_->vehicle_width_m_ / 2.0 + expand_width)));
+  }
+  // next step
+  {
+    double yaw = getYawFromGeometryMsgsQuaternion(next_step_pose.orientation);
+    one_step_move_vehicle_corner_points.push_back(
+      cv::Point2d(
+        next_step_pose.position.x + std::cos(yaw) * (vehicle_info_->wheel_base_m_ + vehicle_info_->front_overhang_m_) -
+        std::sin(yaw) * (vehicle_info_->vehicle_width_m_ / 2.0 + expand_width),
+        next_step_pose.position.y + std::sin(yaw) * (vehicle_info_->wheel_base_m_ + vehicle_info_->front_overhang_m_) +
+        std::cos(yaw) * (vehicle_info_->vehicle_width_m_ / 2.0 + expand_width)));
+    one_step_move_vehicle_corner_points.push_back(
+      cv::Point2d(
+        next_step_pose.position.x + std::cos(yaw) * (vehicle_info_->wheel_base_m_ + vehicle_info_->front_overhang_m_) -
+        std::sin(yaw) * (-vehicle_info_->vehicle_width_m_ / 2.0 - expand_width),
+        next_step_pose.position.y + std::sin(yaw) * (vehicle_info_->wheel_base_m_ + vehicle_info_->front_overhang_m_) +
+        std::cos(yaw) * (-vehicle_info_->vehicle_width_m_ / 2.0 - expand_width)));
+    one_step_move_vehicle_corner_points.push_back(
+      cv::Point2d(
+        next_step_pose.position.x + std::cos(yaw) * (-vehicle_info_->rear_overhang_m_) -
+        std::sin(yaw) * (-vehicle_info_->vehicle_width_m_ / 2.0 - expand_width),
+        next_step_pose.position.y + std::sin(yaw) * (-vehicle_info_->rear_overhang_m_) +
+        std::cos(yaw) * (-vehicle_info_->vehicle_width_m_ / 2.0 - expand_width)));
+    one_step_move_vehicle_corner_points.push_back(
+      cv::Point2d(
+        next_step_pose.position.x + std::cos(yaw) * (-vehicle_info_->rear_overhang_m_) -
+        std::sin(yaw) * (vehicle_info_->vehicle_width_m_ / 2.0 + expand_width),
+        next_step_pose.position.y + std::sin(yaw) * (-vehicle_info_->rear_overhang_m_) +
+        std::cos(yaw) * (vehicle_info_->vehicle_width_m_ / 2.0 + expand_width)));
+  }
+  polygon_ = convexHull(one_step_move_vehicle_corner_points);
+  boost_polygon_ = ConvertToBoostPolygon(polygon_);
+}
+
+inline std::vector<cv::Point2d> OneStepPolygon::convexHull(
+  const std::vector<cv::Point2d> pointcloud)
+{
+  auto centroid = calcCentroid(pointcloud);
+
+  std::vector<cv::Point> normalized_pointcloud;
+  std::vector<cv::Point> normalized_polygon_points;
+  for (size_t i = 0; i < pointcloud.size(); ++i) {
+    normalized_pointcloud.push_back(
+      cv::Point(
+        (pointcloud.at(i).x - centroid.x) * 1000.0, (pointcloud.at(i).y - centroid.y) * 1000.0));
+  }
+  cv::convexHull(normalized_pointcloud, normalized_polygon_points);
```

Comment:
You can replace `cv::convexHull` with `boost::geometry`. If you use `autoware_utils`'s geometry type, you can easily integrate your code with `boost::geometry`.

Reply:
I replaced `cv::convexHull()` with `boost::geometry::convex_hull()`
https://github.com/tier4/obstacle_stop_planner_refine/pull/3/files#diff-c6d1506c16d7dd8e908789d3552fe8646dcf1a9b5cb4ec594f9f2d6173366fe6R80

Reply:
Wow, so nice!

Reply:
Great, but it can be a bit more improved. I'll add comments!